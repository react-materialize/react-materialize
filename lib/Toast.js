'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Button = require('./Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Toast = function (_Component) {
  _inherits(Toast, _Component);

  function Toast(props) {
    _classCallCheck(this, Toast);

    var _this = _possibleConstructorReturn(this, (Toast.__proto__ || Object.getPrototypeOf(Toast)).call(this, props));

    _this.onClick = _this.onClick.bind(_this);
    return _this;
  }

  _createClass(Toast, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          children = _props.children,
          rounded = _props.rounded;

      var classes = { toast: true, rounded: rounded };

      return _react2.default.createElement(
        _Button2.default,
        { onClick: this.onClick, className: (0, _classnames2.default)(classes) },
        children
      );
    }
  }, {
    key: 'onClick',
    value: function onClick() {
      Materialize.toast(this.props.toast, 1000);
    }
  }]);

  return Toast;
}(_react.Component);

Toast.propTypes = {
  toast: _propTypes2.default.string.isRequired,
  children: _propTypes2.default.node,
  rounded: _propTypes2.default.bool
};

exports.default = Toast;