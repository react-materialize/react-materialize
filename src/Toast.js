import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Button from './Button';

class Toast extends Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  render() {
    const { children, rounded, className } = this.props;
    const classes = { toast: true, rounded: rounded };

    return (
      <Button onClick={this.onClick} className={cx(classes, className)}>
        {children}
      </Button>
    );
  }

  onClick() {
    Materialize.toast(this.props.toast, 1000);
  }
}

Toast.propTypes = {
  className: PropTypes.string,
  /**
   * The message to display in the toast
   */
  toast: PropTypes.string.isRequired,
  /**
   * The text to render in the button that fires the toast
   */
  children: PropTypes.node,
  rounded: PropTypes.bool
};

export default Toast;
