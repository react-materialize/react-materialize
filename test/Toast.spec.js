import React from 'react';
import { mount } from 'enzyme';

import Toast from '../src/Toast';
describe('<Toast />', () => {
  const wrappper = mount(<Toast toast="I am a toast">Toast</Toast>);
  window.Materialize = { toast: jest.fn() };

  it('renders properly', () => {
    expect(wrappper).toMatchSnapshot();
  });

  it('should fire the toast on click', () => {
    wrappper.find('button').simulate('click');
    expect(window.Materialize.toast).toHaveBeenCalled();
  });
});
