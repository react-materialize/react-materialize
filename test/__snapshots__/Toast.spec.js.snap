// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> renders properly 1`] = `
<Toast
  toast="I am a toast"
>
  <Button
    className="toast"
    node="button"
    onClick={[Function]}
  >
    <button
      className="btn toast"
      disabled={false}
      onClick={[Function]}
    >
      Toast
    </button>
  </Button>
</Toast>
`;
